server/api.ts

```diff
@@ -1,32 +1,52 @@
-import { Post, User } from './types';
+import { Post, User,Like,Comment } from './types';
 
 // Post APIs
 export interface ListPostsRequest {}
 export interface ListPostsResponse {
   posts: Post[];
 }
-
+export type DeletePostRequest = { postId: string };
+export type DeletePostResponse = {};
 export type CreatePostRequest = Pick<Post, 'title' | 'url' | 'userId'>;
 export interface CreatePostResponse {}
-
-export interface GetPostRequest {}
+export type GetPostRequest = {postId:string}
 export interface GetPostResponse {
   post: Post;
 }
 
 // Comment APIs
+export type CreateCommentRequest = Pick<Comment, 'userId' | 'postId' | 'comment'>;
+export interface CreateCommentResponse {}
+export type GetCommentsRequest = { postId: string };
+export interface GetCommentsResponse {
+  comments: Comment[];
+}
+export type DeleteCommentRequest = { commentId: string };
+export type DeleteCommentResponse = {};
 
 // Like APIs
+export type CreateLikeRequest = Like;
+export interface CreateLikeResponse {}
+export type GetLikesRequest = { postId: string };
+export interface GetLikesResponse {
+  likes: Number;
+}
 
 // User APIs
 export type SignUpRequest = Pick<
   User,
   'email' | 'firstName' | 'lastName' | 'username' | 'password'
 >;
-export interface SignUpResponse {}
```

Comment:
I think you'll want to rebase or merge to adjust your changes, this isn't the latest main. :) looks like you copied the changes manually? this will probably not merge cleanly when you're done with the changes here.